### DISCUSSION TITLE
Quire Commands Not Recognized

### DISCUSSION THREAD

**Discussion by aebancroft** [thegetty/quire#142] [Get Help] [answered]
Hello! I'm trying to run Quire commands in Windows command prompt from the project directory and am getting an error: 

'quire' is not recognized as an internal or external command,
operable program or batch file.

Any suggestions are much appreciated! 

Thank you ---Alex

**Comment by geealbers** [comment, marked as answer]
Hi Alex, that's strange. You'd previously able to run Quire commands before right? Did anything change in your system recently? And if you try a Quire command (say `quire --version`) from a different directory, I assume you get the same message?

**Reply by aebancroft** [reply]
Correct, same error message. I've actually never run Quire commands before, but have been publishing projects successfully. Nothing has changed on my system. Should I try re-cloning the project from Git?

**Reply by geealbers** [reply]
When you say you've "been publishing projects successfully", you mean you're making changes to the Markdown and YAML files in your text editor and pushing them to GitHub, but you've never previewed them on your machine before with the `quire preview` command, correct? If so, the project itself should still be fine and no need to re-clone, but you need to run a fresh install of Quire itself, which is separate from the project. Instructions are here but let me know if it doesn't work or if I misunderstood something: https://quire.getty.edu/documentation/install-uninstall/.

**Comment by aebancroft** [comment]
Thanks, Greg! You were right - looks like Quire needed a fresh install. I can now run Quire commands. 

Now I'm running the quire preview --verbose command and get the following:

C:\Users\ABancroft\museum-essays>quire preview --verbose
Launching preview server
i Navigate to http://localhost:1313 to see your changes.
i Press Control+C or type "quire stop" to stop the preview.
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\webpack.cmd ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn node_modules\\.bin\\webpack.cmd',
  path: 'node_modules\\.bin\\webpack.cmd',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack\\webpack.config.dev.js'
  ]
}

And http://localhost:1313/ site cannot be reached/refused to connect. Any ideas? Would it be better if I start a separate discussion topic for this? 

Thanks, Greg!

**Reply by Erin-Cecele** [reply]
Hi Alex, glad you were able to install Quire successfully. The first time you preview a newly cloned Quire repo you need to run the `quire install` command in your Terminal/Powershell before you can run `quire preview`. This installs Quire's dependencies. You only need to do this once. Let us know if that works!

**Reply by aebancroft** [reply]
Thanks, Erin! A crucial step that I forgot. Preview now works, however it's only showing our cover.md file. How do I see a specific .md file in the content folder? (P.S. this goes back to an internal email you sent me on 3/3/21 re: running quire preview --verbose to identify an issue with our essay not updating. We're having the same problem again and wanted to troubleshoot using the preview.)

**Reply by geealbers** [reply]
@aebancroft, I think you should be able to find a list of essays with links at http://localhost:1313/contents/ but if not, the URLs are constructed based on the names of the .md files minus the file suffix, and any directory they're in. For example, the `awoollett-savery.md` essay, which is in the `paintings` folder would be at a URL of http://localhost:1313/paintings/awoollett-savery/.

**Reply by aebancroft** [reply]
Thanks, Greg! Those URL constructs are good to know. I was able to view all our essays at http://localhost:1313/contents/  except for the one we're trying to troubleshoot. Should be http://localhost:1313/paintings/sbondini-pino/. The essay is pushed to Github - https://github.com/thegetty/museum-essays/blob/stage/content/paintings/sbondini-pino.md. I've checked objects.yml, publication.yml, sbondini-pino.md header; all look ok. Any ideas?

**Reply by aebancroft** [reply]
Jen and I are both working on this essay and have both committed changes to it. We also both recently had to regenerate our Git personal access tokens. Could this have affected our ability to push changes to the museum-essays repo?

**Reply by geealbers** [reply]
I’m afraid I’m out this week so not going to be able to troubleshoot, but can look on Monday. Or if it’s something more urgent, @workergnome could maybe take a look?

**Reply by aebancroft** [reply]
I appreciate you taking a look, Greg. I'm around Monday if it's easier to have a quick Zoom call. Thank you!

**Reply by geealbers** [reply]
The answer to the original query is marked above, but just to close the loop on these other issues here for posterity, and anyone who stumbles onto this post. There were two issues at play, the first was that this particular Quire project was using the `draft` flag on pages, which is [a Hugo feature](https://gohugo.io/variables/page/#page-variables) that's not supported in Quire preview, and on the projects deploy system set up through [Netlify](https://www.netlify.com/), automated builds had been stopped unbeknownst to the project editors.
